src/base/AsyncJobCalls.h

```diff
@@ -176,9 +176,18 @@ JobDialer<Job>::dial(AsyncCall &call)
     } catch (const std::exception &e) {
         debugs(call.debugSection, 3,
                call.name << " threw exception: " << e.what());
+        if (!job) {
+            debugs(call.debugSection, DBG_CRITICAL, "ERROR: Squid BUG: Job invalidated during " <<
+                   call.name << " that threw exception: " << e.what());
```

Comment:
You seem to have forgotten bug 3279. Each and every side effect got reported individually as "The Bug" Just because the number got placed in the logs. I was wading through long debug traces that turned out to be third-party software issues for 8 years.

These exception handlers are even more likely to be hit due to various bugs (unrelated to each other) elsewhere in the code. Resulting in the same hell all over again, but worse.

Reply:
> You seem to have forgotten ...

FWIW, I recommend finding ways to express ideas without negative remarks about collaborators.

> You seem to have forgotten bug 3279. Each and every side effect got reported individually as "The Bug" Just because the number got placed in the logs.

I disagree with the "because the number got placed in the logs" assertion. There are other, IMO much more accurate, explanations for those reports.
 
> These exception handlers are even more likely to be hit due to various bugs (unrelated to each other) elsewhere in the code. Resulting in the same hell all over again, but worse.

I see no connection between "using a bug number in debugs()" and the alleged (but unspecified) hellish effects.

If anything, the number of reports if we use a specific bug number is likely to be _smaller_ than the number of reports if we do not use a specific bug number because, in the latter case, fewer admins would know that this bug is already known to us and, hence, more admins will report them.

AFAICT, the effort to address those reports should not depend in any significant way on whether there is a bug number on that debugs() line.

I consider reports (collected under that bug number) a small but positive step towards fixing the underlying Squid bug(s).